### PR TITLE
Feature/unified media preview

### PR DESCRIPTION
**Pull request by nbradbury**
This PR unifies the fullscreen preview used by the photo picker and the media browser, which required updating the photo picker preview to support remote media. This is in preparation for another PR which will replace `MediaItemFragment` with a fullscreen preview that displays media metadata (filename, size, etc.) in a fading overlay.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaPreviewActivity.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        int width = DisplayUtils.getDisplayPixelWidth(this);
+        int height = DisplayUtils.getDisplayPixelHeight(this);
+
+        if (mediaUri.startsWith("http")) {
+            showProgress(true);
+            String imageUrl = PhotonUtils.getPhotonImageUrl(mediaUri, width, height);
+            mImageLoader.get(imageUrl, new ImageLoader.ImageListener() {
+                @Override
+                public void onResponse(ImageLoader.ImageContainer response, boolean isImmediate) {
+                    if (response.getBitmap() != null) {
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaPreviewActivity.java]: Should we add a call to ` if (isFinishing()) return;` here?

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaPreviewActivity.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                        showProgress(false);
+                        mImageView.setImageBitmap(response.getBitmap());
+                    }
+                }
+                @Override
+                public void onErrorResponse(VolleyError error) {
+                    AppLog.e(AppLog.T.MEDIA, error);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaPreviewActivity.java]: Same as above, should we check for  `isFinishing()` ?

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaPreviewActivity.java (hunk 1 above)
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaPreviewActivity.java]: What will happen in the unlikely event that `getBitmap` returns `null` here?
Should we dismiss the progress bar and show the error?

**Comment by nbradbury** [comment]
> What will happen in the unlikely event that getBitmap returns null here?

As [this Stackoverflow comment](http://stackoverflow.com/a/28553767/1673548) explains, a `null` bitmap is a normal immediate response so we should do nothing special.

> should we check for isFinishing() ?

Added in 8c430cc - ready for another round!

**Comment by daniloercoli** [comment]
🙇

**Pull request merged**